python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -360,11 +477,26 @@ def build(self, input_shape):
             key_dtype=tf.string,
             value_dtype=tf.int64,
         )
-        self.lookup_table = lookup.StaticVocabularyTable(
-            initializer=table_init,
-            num_oov_buckets=self.num_oov_buckets,
-            name="{}_lookup_table".format(self.feature_name),
-        )
+
+        """
+        NOTE:
```

Comment:
+1
Could we add:
1. A global doc above NOTE (about what build does)
2. Why do we choose to build a StaticVocabularyTable or StaticHashTable depending on the num_oov_buckets?

Reply:
Because StaticHashTable doesn't have num_oov_buckets argument and StaticVocabularyTable does not have default_value argument. For most cases, StaticVocabularyTable is sufficient. But when we want to use a custom default_value(like in the case of the dropout function), we need to use StaticHashTable.

Reply:
Added the doc above note regarding what build does.

Reply:
We should also add a form of what you just said for the context as doc:
_Because StaticHashTable doesn't have num_oov_buckets argument and StaticVocabularyTable does not have default_value argument. For most cases, StaticVocabularyTable is sufficient. But when we want to use a custom default_value(like in the case of the dropout function), we need to use StaticHashTable._
Mostly adding the _why_ :)

Reply:
ie that we need it for instance to add a custom default_value, like in the case of the dropout function.